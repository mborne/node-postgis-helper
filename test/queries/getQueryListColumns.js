const expect = require("chai").expect;

const queries = require('../../src/queries');

describe("Test queries.getQueryListColumns...", function () {

    it("should return an expected query for queries.getQueryListTables('the_schema','the_table')", function () {
        let query = queries.getQueryListColumns('the_schema','the_table');
        //console.log(JSON.stringify(query,null,2));
        let expected = require('./expected/getQueryListColumns.json');
        expect(query).to.deep.equals(expected);
    });


});

