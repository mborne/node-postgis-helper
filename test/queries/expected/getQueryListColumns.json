{
    "text": "SELECT\n    a.attname as name,\n    format_type(a.atttypid, a.atttypmod) AS type,\n    (a.attnotnull = false) as is_nullable\nfrom pg_attribute a\n    where a.attrelid = ($1 ||'.' || $2)::regclass\n    and a.attnum > 0\n    and not a.attisdropped",
    "values": [
        "the_schema",
        "the_table"
    ]
}